### ISSUE TITLE
`type.maybe` suggests Option type, but is more like the "null object" pattern?

### ISSUE THREAD

**Issue by josh-bitovi** [canjs/can-type#6]
Small naming nit :)

Maybe `type.nullable` is more accurate?

Original discussion: https://bitovi-community.slack.com/archives/CFC22NZ8A/p1560447176065900

Related reading:
- https://en.wikipedia.org/wiki/Option_type
- https://en.wikipedia.org/wiki/Nullable_type#Compared_with_option_types

**Comment by justinbmeyer** [comment]
Does `nullable` also usually include `undefined`?  I was afraid that `nullable` would leave folks confused when `undefined` also was allowed.

**Comment by josh-bitovi** [comment]
Yeah that's tricky. `nullOrUndefinedOr(Type)` doesn't read very well. `blank` is usually used to indicate empty (or whitespace only or both) strings... hmm. I'm not sure. Maybe lodash has something to refer to null and undefined?

**Comment by josh-bitovi** [comment]
@rlmcneary2 says "the industry" is heading toward using `nullish` as a term for `null`/`undefined`/`Some(a)`.

**Comment by josh-bitovi** [comment]
@justinbmeyer sgty?

**Comment by rlmcneary2** [comment]
@josh-bitovi I think I just interjected "nullish" without any real thought whatsoever. Industry be damned!

**Comment by josh-bitovi** [comment]
Oh I misheard then.. yeah it's kind of like "blank"... but I don't really like that either because of whitespace-only & empty strings.

**Comment by justinbmeyer** [comment]
`.missingOr(String)`?

**Comment by justinbmeyer** [comment]
I like `type.nullish( Boolean )` now

**Comment by rjgotten** [comment]
> "the industry" is heading toward using nullish as a term for null/undefined

As a means to indicate values that are null-like. Not as a means of indicating data-types that allow nullish values. For those the colloquial term is still "nullable", heralding from the C# world.

> @justinbmeyer 
> I like `type.nullish( Boolean )` now

You really should be using `type.nullable( Boolean )` if you want to avoid conflating principles.
